source/extensions/filters/network/dubbo_proxy/router/router_impl.cc

```diff
@@ -0,0 +1,345 @@
+#include "extensions/filters/network/dubbo_proxy/router/router_impl.h"
+
+#include "envoy/config/filter/network/dubbo_proxy/v2alpha1/dubbo_proxy.pb.h"
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/thread_local_cluster.h"
+
+#include "extensions/filters/network/dubbo_proxy/app_exception.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+namespace Router {
+
+void Router::onDestroy() {
+  if (upstream_request_) {
+    upstream_request_->resetStream();
+  }
+  cleanup();
+}
+
+void Router::setDecoderFilterCallbacks(DubboFilters::DecoderFilterCallbacks& callbacks) {
+  callbacks_ = &callbacks;
+}
+
+Network::FilterStatus Router::transportBegin() {
+  upstream_request_buffer_.drain(upstream_request_buffer_.length());
+  ProtocolDataPassthroughConverter::initProtocolConverter(upstream_request_buffer_);
+  return Network::FilterStatus::Continue;
+}
+
+Network::FilterStatus Router::transportEnd() {
+  ASSERT(upstream_request_);
+
+  upstream_request_->encodeData(upstream_request_buffer_, true);
+
+  if (upstream_request_->metadata_->message_type() == MessageType::Oneway) {
+    // No response expected
+    upstream_request_->onResponseComplete();
+    cleanup();
```

Comment:
Remove ActiveMessage?

Reply:
If the request is one_way, the current stream is released after the Filter ends the callback, and there is no need to call the resetStream function.